src/impl/tpls/KokkosBlas_gesv_tpl_spec_decl.hpp

```diff
@@ -44,6 +44,216 @@
+     Kokkos::View<double**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+                  Kokkos::MemoryTraits<Kokkos::Unmanaged> >, \
+     true, ETI_SPEC_AVAIL> { \
+  typedef double SCALAR; \
+  typedef Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+      Kokkos::MemoryTraits<Kokkos::Unmanaged> > AViewType; \
+  typedef Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+      Kokkos::MemoryTraits<Kokkos::Unmanaged> > BViewType; \
+ \
+  static void \
+  gesv (const char pivot[], \
+        const AViewType& A, \
+        const BViewType& B) { \
+    \
+    Kokkos::Profiling::pushRegion("KokkosBlas::gesv[TPL_BLAS,double]"); \
+    gesv_print_specialization<AViewType,BViewType>(); \
+    const bool pivot_t = (pivot[0]=='Y') || (pivot[0]=='y'); \
+    \
+    const int N    = static_cast<int> (A.extent(1)); \
+    const int AST  = static_cast<int> (A.stride(1)); \
+    const int LDA  = (AST == 0) ? 1 : AST; \
+    const int BST  = static_cast<int> (B.stride(1)); \
+    const int LDB  = (BST == 0) ? 1 : BST; \
+    const int NRHS = static_cast<int> (B.extent(1)); \
+    \
+    int info = 0; \
+    \
+    if(pivot_t) { \
+      int *ipiv = NULL; \
+      ipiv = (int*)malloc(N*sizeof(int)); \
+      HostBlas<double>::gesv (N, NRHS, A.data(), LDA, ipiv, B.data(), LDB, info); \
+      free( ipiv ); \
+    } \
+    Kokkos::Profiling::popRegion(); \
+  } \
+};
+
+#define KOKKOSBLAS_SGESV_BLAS( LAYOUT, MEM_SPACE, ETI_SPEC_AVAIL ) \
+template<class ExecSpace> \
+struct GESV< \
+     Kokkos::View<float**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+                  Kokkos::MemoryTraits<Kokkos::Unmanaged> >, \
+     Kokkos::View<float**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+                  Kokkos::MemoryTraits<Kokkos::Unmanaged> >, \
+     true, ETI_SPEC_AVAIL> { \
+  typedef float SCALAR; \
+  typedef Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+      Kokkos::MemoryTraits<Kokkos::Unmanaged> > AViewType; \
+  typedef Kokkos::View<SCALAR**, LAYOUT, Kokkos::Device<ExecSpace, MEM_SPACE>, \
+      Kokkos::MemoryTraits<Kokkos::Unmanaged> > BViewType; \
+      \
+  static void \
+  gesv (const char pivot[], \
+        const AViewType& A, \
+        const BViewType& B) { \
+    \
+    Kokkos::Profiling::pushRegion("KokkosBlas::gesv[TPL_BLAS,float]"); \
+    gesv_print_specialization<AViewType,BViewType>(); \
+    const bool pivot_t = (pivot[0]=='Y') || (pivot[0]=='y'); \
+    \
+    const int N    = static_cast<int> (A.extent(1)); \
+    const int AST  = static_cast<int> (A.stride(1)); \
+    const int LDA  = (AST == 0) ? 1 : AST; \
+    const int BST  = static_cast<int> (B.stride(1)); \
+    const int LDB  = (BST == 0) ? 1 : BST; \
+    const int NRHS = static_cast<int> (B.extent(1)); \
+    \
+    int info = 0; \
+    \
+    if(pivot_t) { \
+      int *ipiv = NULL; \
+      ipiv = (int*)malloc(N*sizeof(int)); \
```

Comment:
This is C++; it's preferred to use `new` / `delete` (or better yet, `unique_ptr<int[]>`).

Reply:
@mhoemmen Thanks. ```new```/```delete``` are now used.